base/iobuffer.jl

```diff
@@ -1,26 +1,101 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-## work with AbstractVector{UInt8} via I/O primitives ##
+# IOBuffer is a Memory{UInt8} backed IO type for in-memory IO.
+
+# Here, u represents used bytes (already read), X represents bytes still to read,
+# - represents bytes uninitialized data but which can be written to later.
+
+#   uuuuuuuuuuuuuXXXXXXXXXXXXX------------
+#   |       |    |           |           |    |
+#   |       |    ptr         size        |    maxsize (≥ lastindex)
+#   1       mark (zero-indexed)          lastindex(data)
+
+#            AFTER COMPACTION
+# Mark, ptr and size decreases by `mark`
+
+#   uuuuuXXXXXXXXXXXXX---------------------
+#  ||    |           |                    |    |
+#  |1    ptr         size                 |    maxsize (≥ lastindex)
+#  mark (zero-indexed)                    lastindex(data)
+
+# * The underlying array is always 1-indexed
+# * The IOBuffer has full control (ownership) of the underlying array.
+# * Data in 1:mark can be deleted, shifting the whole thing to the left
+#   to make room for more data, without replacing or resizing data
+
+# Internal trait object used to access unsafe constructors.
+struct Unsafe end
+const unsafe = Unsafe()
```

Comment:
These names feel a bit too generic to put into the `Base` namespace.

Reply:
Changed the name to `UnsafeMethod` and `unsafe_method`. They can be called something arbitrary, but I do think it's nice that they're available for all internal, unsafe methods.